src/pages/api/selectSalesOfferPackage.ts

```diff
@@ -0,0 +1,87 @@
+    getPeriodGeoZoneTicketJson,
+    getPeriodMultipleServicesTicketJson,
+    getFlatFareTicketJson,
+    unescapeAndDecodeCookie,
+} from './apiUtils';
+import { isSessionValid } from './service/validator';
+import { SALES_OFFER_PACKAGES_ATTRIBUTE, FARE_TYPE_COOKIE, PERIOD_TYPE_COOKIE } from '../../constants';
+import { NextApiRequestWithSession, SelectSalesOfferPackageWithError } from '../../interfaces';
+import { updateSessionAttribute } from '../../utils/sessions';
+
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        if (!isSessionValid(req, res)) {
+            throw new Error('Session is invalid.');
+        }
+
+        const cookies = new Cookies(req, res);
+        const fareTypeCookie = unescapeAndDecodeCookie(cookies, FARE_TYPE_COOKIE);
+        const fareTypeObject = JSON.parse(fareTypeCookie);
+        const { fareType } = fareTypeObject;
+
+        if (!req.body || Object.keys(req.body).length === 0) {
+            const salesOfferPackagesAttributeError: SelectSalesOfferPackageWithError = {
+                errorMessage: 'Choose at least one service from the options',
+            };
+            updateSessionAttribute(req, SALES_OFFER_PACKAGES_ATTRIBUTE, salesOfferPackagesAttributeError);
+            redirectTo(res, `/selectSalesOfferPackage`);
+            return;
+        }
+
+        const uuid = getUuidFromCookie(req, res);
+
+        let userDataJson;
+
+        if (fareType === 'single') {
+            userDataJson = getSingleTicketJson(req, res);
+            await putUserDataInS3(userDataJson, uuid);
+            redirectTo(res, '/thankyou');
+            return;
+        }
+        if (fareType === 'return') {
+            userDataJson = getReturnTicketJson(req, res);
+            await putUserDataInS3(userDataJson, uuid);
+            redirectTo(res, '/thankyou');
+            return;
+        }
+        if (fareType === 'period') {
+            const periodTypeCookie = unescapeAndDecodeCookie(cookies, PERIOD_TYPE_COOKIE);
+            const periodTypeObject = JSON.parse(periodTypeCookie);
+            const { periodTypeName } = periodTypeObject;
+
+            if (periodTypeName === 'periodGeoZone') {
+                userDataJson = getPeriodGeoZoneTicketJson(req, res);
+                await putUserDataInS3(await userDataJson, uuid);
+                redirectTo(res, '/thankyou');
+                return;
+            }
+            if (periodTypeName === 'periodMultipleServices') {
+                userDataJson = getPeriodMultipleServicesTicketJson(req, res);
+                await putUserDataInS3(userDataJson, uuid);
+                redirectTo(res, '/thankyou');
+                return;
+            }
+        }
+        if (fareType === 'flatFare') {
+            userDataJson = getFlatFareTicketJson(req, res);
+            await putUserDataInS3(userDataJson, uuid);
+            redirectTo(res, '/thankyou');
+            return;
+        }
```

Comment:
We could make this block of if statements a bit shorter and more readable by only setting `userDataJson` inside - by removing the below from each if block and have it just once at the end
```
await putUserDataInS3(userDataJson, uuid);
            redirectTo(res, '/thankyou');
            return;
```
That way we would replace 15 lines of code with 3